phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1451,7 +1454,35 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
         }
         return null; // will never make it here
     }
-    
+
+    private boolean isSplitPolicyUpdatedForTaskTable(
+            final TableDescriptorBuilder tdBuilder) {
+        boolean isTaskTable = false;
+        String taskTable = PhoenixDatabaseMetaData.SYSTEM_TASK_NAME;
+        if (tdBuilder.build().getTableName().getNameAsString()
+                .equals(taskTable)) {
+            isTaskTable = true;
+        } else {
```

Comment:
Can't we use the actual namespace mapping property to get the exact expected name i.e. SYSTEM.TASK vs SYSTEM:TASK instead of trying both variations?